### ISSUE TITLE
tables, h3 and h4 rendering is broken

### ISSUE THREAD

**Issue by tkachuk-labs** [sotte/presenting.vim#34]
This markdown
```markdown
# Tables

| Last Name      | First Name | Born | Bacon Number |
|----------------|------------|------|--------------|
| Bacon          | Kevin      | 1958 | 0            |
| Chan           | Jackie     | 1954 | 2            |
| Reagan         | Ronald     | 1911 | 2            |
| Ryan           | Meg        | 1961 | 1            |
| Saldaña        | Zoë        | 1978 | 2            |
| Schwarzenegger | Arnold     | 1947 | 2            |
| Temple         | Shirley    | 1928 | 3            |
```
Renders as
```
                                                                          _____     _     _           
                                                                         |_   _|_ _| |__ | | ___  ___ 
                                                                           | |/ _` | '_ \| |/ _ \/ __|
                                                                           | | (_| | |_) | |  __/\__ \
                                                                           |_|\__,_|_.__/|_|\___||___/
                                                                                                      

«tr»                                                            ┏━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━┳━━━━━━┳━━━━━━━━━━━━━━┓
«th»                                                            ┃ Last Name      ┃ First Name ┃ Born ┃ Bacon Number ┃
«tr»                                                            ┣━━━━━━━━━━━━━━━━╋━━━━━━━━━━━━╋━━━━━━╋━━━━━━━━━━━━━━┫
«tr»                                                            ┃ Bacon          ┃ Kevin      ┃ 1958 ┃ 0            ┃
«tr»                                                            ┃ Chan           ┃ Jackie     ┃ 1954 ┃ 2            ┃
«tr»                                                            ┃ Reagan         ┃ Ronald     ┃ 1911 ┃ 2            ┃
«tr»                                                            ┃ Ryan           ┃ Meg        ┃ 1961 ┃ 1            ┃
«tr»                                                            ┃ Saldaña        ┃ Zoë        ┃ 1978 ┃ 2            ┃
«tr»                                                            ┃ Schwarzenegger ┃ Arnold     ┃ 1947 ┃ 2            ┃
«tr»                                                            ┃ Temple         ┃ Shirley    ┃ 1928 ┃ 3            ┃
«tr»                                                            ┗━━━━━━━━━━━━━━━━┻━━━━━━━━━━━━┻━━━━━━┻━━━━━━━━━━━━━━┛

```

**Comment by chris542** [comment]
I also see this issue :(
![image](https://user-images.githubusercontent.com/17423698/112772507-b405ce80-908d-11eb-8ae3-f9ade2317e98.png)

**Comment by rohitvvv** [comment]
How to get rid of the tags that render on the left?

**Comment by sotte** [comment]
#26 introduced the much nicer rendering of presentations (thanks again!), but it's not as robust as it should have been (my bad, I should have checked more).

I'll have some time this weekend and will check it out.

**Comment by f1se4** [comment]
Hi,

I have cloned this respository to be able to be used in ‘ivim’ ios version of vim. By the way I have solved (or it works for me) tags issues with tables and headers.

Here you have the repository: https://github.com/f1se4/presenting_ivim.vim.git

And, really really love this. In an ipad-air where I’m using this it’s perfect way to have ‘geek’ presentation done in the best editor in less than any other tool I could imagine. Thank you,

**Comment by chris542** [comment]
hey @sotte , has this been resolved? I just tried out the latest version but it still has the same issue :(

**Comment by sotte** [comment]
@chris542 well, life happened :)

I did some investigation. The conceal feature does not work as intended.

The first conceal on a page/slide always works. It can be any heading or table!
E.g. if you have a md file like this the rendering/concealing works.
```
| Last Name      | First Name | Born | Bacon Number |
|----------------|------------|------|--------------|
| Bacon          | Kevin      | 1958 | 0            |
| Chan           | Jackie     | 1954 | 2            |
```

Did some research on `conceal`, but did not see anything obvious wrong. I'll have to investigate further.

(I also found some inconsistencies how figlet and not figlet headings are handled. I'll address this separately.)

**Comment by sotte** [comment]
I checked out many versions and at least for my current setup (nvim v0.5) this feature never worked. I'm sure it did at some point though. 

@PhilRunninger you implemented this feature. Can you check if conceal still works on your end? That would be great!

```
❯ nvim --version
NVIM v0.5.0-dev+1369-g64bdaef32
Build type: Release
LuaJIT 2.1.0-beta3
```

**Comment by f1se4** [comment]
I don't know if it helps, but for me has worked by commenting line 26 from /autoload/markdown.vim 

let l:text = s:Center(l:text, '<<tr>>')

thks

**Comment by sotte** [comment]
Thanks @f1se4. I assume you're using the non-figlet version?

@PhilRunninger introduced the markers (tr, h1, etc.) to, well, mark a given line and assign/[link](http://vimdoc.sourceforge.net/htmldoc/syntax.html#:highlight-link) a normal syntax element to the newly marked one. Removing the `<<tr>>` does the trick (nothing to conceal when it does not exists), but it's only a workaround. I'll keep looking for a proper solution.

Thanks again!

**Comment by f1se4** [comment]
yaps, well the python-figlet version hahahahhaa 😄

**Comment by PhilRunninger** [comment]
@sotte 

I can confirm it is working as advertised on Windows 10 with these versions of `nvim` and `nvim-qt`
```
NVIM v0.5.0-828-g0a95549d6
Build type: RelWithDebInfo
LuaJIT 2.1.0-beta3
```
and
```
NVIM v0.5.0-dev+1372-g056c464e8
Build type: RelWithDebInfo
LuaJIT 2.1.0-beta3
```

On my Macbook with `nvim` version
```
NVIM v0.4.4
Build type: Release
LuaJIT 2.0.5
```
the tables are working correctly, but this issue may be showing up in the code block where the `` ```vim `` code fence is not being concealed. Never mind; this is caused by the older `syntax/markdown.vim` file that ships with the v0.4.4 version. This was fixed in this commit: https://github.com/tpope/vim-markdown/commit/c043a93c847dbaba96aac643df818baafa04a3e5.

In summary, I'd say it's working as intended, and as it always has for me. There could be some obscure setting that I have in my [config](https://github.com/PhilRunninger/nvim_config) that makes it all work, so feel free to look around in it.

I'll be AFK for a while, but I'll check in when I can.

**Comment by sotte** [comment]
@PhilRunninger thanks, that is super helpful!

I'm using https://github.com/plasticboy/vim-markdown, not `tpope/markdown-vim` (but I did in the past). I just tested with tpope's version and it looks much better:

Raw md:

```
# presenting.vim
### maintainer: Stefan Otte

#### a presentation tool for vim

<!--
  Comments are removed. They must be
  in the form of an HTML comment tag.
-->

\```python
def foo(a: str) -> None:
    return a.upper()
\```

| Last Name      | First Name | Born | Bacon Number |
|----------------|------------|------|--------------|
| Bacon          | Kevin      | 1958 | 0            |

- [ ] Not done yet
  - [ ] Task 1
- [x] Done
  - [x] Task 1
```

Rendered in vim with `tpope/markdown-vim`:
![image](https://user-images.githubusercontent.com/79138/120761619-af3e1a80-c515-11eb-86d9-b626086837f0.png)

Rendered with presenting:
![image](https://user-images.githubusercontent.com/79138/120761322-5d959000-c515-11eb-80e7-2be1984218a1.png)

I get syntax hl in the raw md, but not in presenting.vim (I had to enable fenced code blocks `let g:markdown_fenced_languages = ["python"]`).

If this behavior is due to incompatibility between these two md plugins I'm not quite sure how to guarantee that presenting.vim works with both (without too much effort at least).

**Comment by sotte** [comment]
Another try with no specific md plugin installed...and it woks :) fooo...
![image](https://user-images.githubusercontent.com/79138/120762982-17413080-c517-11eb-8aea-03ebc332b3a1.png)

**Comment by sotte** [comment]
@chris542 @rohitvvv @tkachuk-labs
- What is your vim version?
- What markdown plugin are you using (if any)?
- What are your settings for the markdown plugin?

**Comment by rohitvvv** [comment]
NVIM v0.5.0-dev+1087-g5ad32885d
Build type: Release
LuaJIT 2.1.0-beta3

I deleted the vim-markdown from tpope. 
fadein/vim-FIGlet

**Comment by chris542** [comment]
> @chris542 @rohitvvv @tkachuk-labs
> 
> - What is your vim version?
> 
> - What markdown plugin are you using (if any)?
> 
> - What are your settings for the markdown plugin?

I'm using the latest vim8

These are the only configs i have for markdown

"sotte/presenting.vim
let g:presenting_figlets = 1

"gabrielelana/vim-markdown
let g:markdown_enable_mappings = 1
let g:markdown_enable_spell_checking = 0
let g:markdown_enable_conceal = 1
let g:markdown_enable_folding = 1
let g:vim_markdown_conceal_code_blocks = 0
let g:vim_markdown_fenced_languages = ['csharp=cs', 'js=javascript', 'sh=bash']
